### PR TITLE
add use_nms option

### PR DESCRIPTION
**Pull request by wusize**
Add use_nms option

### CODE REVIEW

File: mmpose/datasets/datasets/top_down/topdown_aic_dataset.py [hunk 1]
```diff
@@ -55,6 +55,10 @@ def __init__(self,
         self.bbox_file = data_cfg['bbox_file']
         self.image_thr = data_cfg['image_thr']
 
+        if 'use_nms' in data_cfg:
```
Comment (innerlee) [inline mmpose/datasets/datasets/top_down/topdown_aic_dataset.py:58]: `data_cfg.get('use_nms', True)

Comment (jin-s13) [reply mmpose/datasets/datasets/top_down/topdown_aic_dataset.py]: fixed.

File: mmpose/datasets/datasets/top_down/topdown_coco_dataset.py [hunk 2]
```diff
@@ -361,15 +361,18 @@ def evaluate(self, outputs, res_folder, metric='mAP', **kwargs):
                 # rescoring
                 n_p['score'] = kpt_score * box_score
 
-            nms = soft_oks_nms if self.soft_nms else oks_nms
-            keep = nms(list(img_kpts), oks_thr, sigmas=self.sigmas)
+            if self.use_nms:
```
Comment (innerlee) [inline mmpose/datasets/datasets/top_down/topdown_coco_dataset.py:364]: `if self.use_nms and len(img_kpts) > 0:`

Comment (innerlee) [reply mmpose/datasets/datasets/top_down/topdown_coco_dataset.py:364]: According to discussion, nms will always return values if input is non-empty

Comment (wusize) [reply mmpose/datasets/datasets/top_down/topdown_coco_dataset.py:364]: We might need these lines in case of len(img_kpts)=0.

Comment (innerlee) [reply mmpose/datasets/datasets/top_down/topdown_coco_dataset.py:364]: see the code posted above

Comment (wusize) [reply mmpose/datasets/datasets/top_down/topdown_coco_dataset.py:364]: Em,,, I mean it's very likely that **"img_kpts = kpts[image_id]"** gets an empty list [ ], in the case of which nms will return an empty list [ ] and **"if len(keep) == 0: valid_kpts.append(img_kpts)"** would be necessary. 
2020-11-29 23-47-14屏幕截图

Comment (innerlee) [reply mmpose/datasets/datasets/top_down/topdown_coco_dataset.py:364]: Okay, so nms does not guarantee that the return be non-empty when img_kpts is non-empty?

Comment (wusize) [reply mmpose/datasets/datasets/top_down/topdown_coco_dataset.py:364]: Oh, now I get the point. I erroneously thought that **[img_kpts[_keep] for _keep in keep]** would raise raise if **keep=[]** and **img_kpts=[]**, but it won't.
2020-11-30 08-28-08屏幕截图
And sure nms does guarantee that. I was talking about the empty case but now it's also not a problem. So **if len(keep) == 0: valid_kpts.append(img_kpts)** is unnecessary in both cases. I will rm these lines.

**Review by innerlee — APPROVED** [review]

**Pull request merged**